src/frontend/web_application/src/components/ContactDetails/index.jsx

```diff
@@ -225,19 +304,86 @@ class ContactDetails extends Component {
     );
   }
 
-  renderContactDetailsForm() {
+  renderFormSelector = () => {
     const { __, contact } = this.props;
+
+    const emailOption = {
+      name: __('contact.form-selector.email_form.label'),
+      obj: (<EmailForm onSubmit={this.makeHandleAddContactDetail('emails')} __={__} />),
+    };
+
+    const phoneOption = {
+      name: __('contact.form-selector.phone_form.label'),
+      obj: (<PhoneForm onSubmit={this.makeHandleAddContactDetail('phones')} __={__} />),
+    };
+
+    const imOption = {
+      name: __('contact.form-selector.im_form.label'),
+      obj: (<ImForm onSubmit={this.makeHandleAddContactDetail('ims')} __={__} />),
+    };
+
+    const addressOption = {
+      name: __('contact.form-selector.address_form.label'),
+      obj: (<AddressForm onSubmit={this.makeHandleAddContactDetail('addresses')} __={__} />),
+    };
+
+    const formsOptions = [
+      { name: ' - ', obj: null },
+      !contact.emails ? emailOption : null,
+      !contact.phones ? phoneOption : null,
+      !contact.ims ? imOption : null,
+      !contact.addresses ? addressOption : null,
+    ].filter(option => option !== null); // only return new forms for empty contact's attributes
+
+    return (
+      <FormSelector __={__} formsOptions={formsOptions} />
+    );
+  }
+
+  renderAddFormButton = (obj) => {
+    const { __ } = this.props;
+
+    return (
+      <ItemContent large>
+        <FormButton obj={obj} __={__} />
```

Comment:
you should pass it via `children` property it would be more conventional cf. the component